File: cpp/cpp_1000-/1051.cc
// C++
// leetcode 1051
// https://leetcode.cn/problems/height-checker/

class Solution {
public:
    int heightChecker(vector<int>& heights) {
        vector<int> expected(heights.begin(), heights.end());
        sort(expected.begin(), expected.end());
        int n = heights.size(), res = 0;
        for(int i = 0; i < n; ++i) {
            if(heights[i] != expected[i])
                res++;
        }
        return res;
    }
};

File: Go/Go_1000-/1051.go
// Go
// leetcode 1051
// https://leetcode.cn/problems/height-checker/

func heightChecker(heights []int) int {
	n := len(heights)
	arr := make([]int, n)
	copy(arr, heights)
	sort.Ints(arr)
	res := 0
	for i := 0; i < n; i++ {
		if arr[i] != heights[i] {
			res++
		}
	}
	return res
}